src/backend/commands/extension.c

```diff
@@ -862,7 +863,7 @@ execute_extension_script(Node *stmt,
 
 	(void) set_config_option("search_path", pathbuf.data,
 							 PGC_USERSET, PGC_S_SESSION,
-							 true, true);
+							 GUC_ACTION_SAVE, true);
```

Comment:
What was the reason of adding GUC_ACTION_SAVE guc here?

Reply:
This flag is a direct port of 6x's logic. gpdb5 doesn't save nested gucs correctly without this flag. Moreover, it doesn't have any tests checking its correctness. It was fixed and test suit was added

Reply:
1) Why this placed in commit named "Fix compilation warnings"?
2) Where is any explanation (PR message, commit message) of this?

Reply:
Description added

Reply:
Description added